### ISSUE TITLE
Error handling/ HTTP status codes

### ISSUE THREAD

**Issue by damb** [EIDA/mediatorws#21]
Implement a proper *Flask*/*Flask-Restful*/*webargs* error handling. Follow the 
*Common HTTP status codes returned by FDSN services* (https://www.fdsn.org/webservices/FDSN-WS-Specifications-1.1.pdf)

**Comment by damb** [comment]
Solving this issue also implies returning the correct `nodata` value parameter and Content-Type, i.e.
```
$ curl -v "http://mediator-devel.ethz.ch/fdsnws/dataselect/1/query?net=CH&sta=BALST&loc=--&cha=HHZ&starttime=1970-02-27T06:30:00&endtime=1970-02-27T06:35:00&nodata=404"
*   Trying 129.132.144.211...
* TCP_NODELAY set
* Connected to mediator-devel.ethz.ch (129.132.144.211) port 80 (#0)
> GET /fdsnws/dataselect/1/query?net=CH&sta=BALST&loc=--&cha=HHZ&starttime=1970-02-27T06:30:00&endtime=1970-02-27T06:35:00&nodata=404 HTTP/1.1
> Host: mediator-devel.ethz.ch
> User-Agent: curl/7.55.1
> Accept: */*
> 
< HTTP/1.1 204 NO CONTENT
< Date: Tue, 27 Mar 2018 10:02:51 GMT
< Server: Apache/2.4.18 (Ubuntu)
< Content-Type: application/json
< 
* Connection #0 to host mediator-devel.ethz.ch left intact
```
versus
```
$ curl -v "http://eida.ethz.ch/fdsnws/dataselect/1/query?net=CH&sta=BALST&loc=--&cha=HHZ&starttime=1970-02-27T06:30:00&endtime=1970-02-27T06:35:00&nodata=404"
*   Trying 129.132.144.217...
* TCP_NODELAY set
* Connected to eida.ethz.ch (129.132.144.217) port 80 (#0)
> GET /fdsnws/dataselect/1/query?net=CH&sta=BALST&loc=--&cha=HHZ&starttime=1970-02-27T06:30:00&endtime=1970-02-27T06:35:00&nodata=404 HTTP/1.1
> Host: eida.ethz.ch
> User-Agent: curl/7.55.1
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< Date: Tue, 27 Mar 2018 10:02:24 GMT
< Server: SeisComP3-FDSNWS/1.1.1
< Access-Control-Expose-Headers: WWW-Authenticate
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: Authorization
< Content-Type: text/plain
< Content-Length: 0
< 
* Connection #0 to host eida.ethz.ch left intact
```

**Comment by kaestli** [comment]
Note that the unexpected response type header (application/json) is retuned only in case of a nodata situation.

**Comment by damb** [comment]
Implemented with 77cfb2a. Found a pretty neat solution using parameterized decorators :). In addition, I immediately implemented the FDSNWS conform error handling for `stationlite`. Closed.
